rust/kcl-lib/src/std/array.rs

```
@@ -257,6 +260,35 @@ async fn call_reduce_closure(
     Ok(out)
 }
 
+pub async fn push(exec_state: &mut ExecState, args: Args) -> Result<KclValue, KclError> {
+    let array = args.get_unlabeled_kw_arg_typed(
+        "array",
+        &RuntimeType::Array(Box::new(RuntimeType::any()), ArrayLen::None),
+        exec_state,
+    )?;
+    let item: KclValue = args.get_kw_arg("item")?;
+
+    let KclValue::HomArray { value: values, ty } = array else {
+        let meta = vec![args.source_range];
+        let actual_type = array.human_friendly_type();
+        return Err(KclError::Semantic(KclErrorDetails {
+            source_ranges: meta,
+            message: format!("You can't push to a value of type {actual_type}, only an array"),
+        }));
+    };
+    let ty = if item.has_type(&ty) {
+        ty
+    } else {
+        // TODO: The user pushed an item of a different type than the array.
+        // What should happen here?
+        RuntimeType::any()
```

Should we throw a runtime error here?

---

No, I think we need parametric polymorphism.